concrete/src/Page/Page.php

```diff
@@ -2256,14 +2256,23 @@ public function addBlock($bt, $a, $data)
         $theme = $this->getCollectionThemeObject();
         if ($btHandle && $theme) {
             $areaTemplates = [];
+            $pageTypeTemplates = [];
             if (is_object($a)) {
                 $areaTemplates = $a->getAreaCustomTemplates();
             }
             $themeTemplates = $theme->getThemeDefaultBlockTemplates();
             if (!is_array($themeTemplates)) {
                 $themeTemplates = [];
+            } else {
+                foreach($themeTemplates as $key => $template){
+                    $pt = ($this->getPageTypeHandle()) ? $this->getPageTypeHandle() : 'page';
+                    if(is_array($template) && $key == $pt){
+                        $pageTypeTemplates = $template;
+                    }
+                    unset($themeTemplates[$key]);
```

Comment:
This unsets $themeTemplates[$key] although $pageTypeTemplates might not have been set. 
https://github.com/concrete5/concrete5/pull/6456#issuecomment-385443297

Reply:
sorry, have seen that this is already  fixed in c5-project.